optimize/backend/src/test/java/io/camunda/optimize/service/OutlierAnalysisServiceTest.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.optimize.service;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThatThrownBy;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import io.camunda.optimize.dto.optimize.DefinitionType;
+import io.camunda.optimize.dto.optimize.query.analysis.DurationChartEntryDto;
+import io.camunda.optimize.dto.optimize.query.analysis.FindingsDto;
+import io.camunda.optimize.dto.optimize.query.analysis.FlowNodeOutlierParametersDto;
+import io.camunda.optimize.dto.optimize.query.analysis.OutlierAnalysisServiceParameters;
+import io.camunda.optimize.dto.optimize.query.analysis.ProcessDefinitionParametersDto;
+import io.camunda.optimize.service.db.reader.DurationOutliersReader;
+import io.camunda.optimize.service.security.util.definition.DataSourceDefinitionAuthorizationService;
+import jakarta.ws.rs.ForbiddenException;
+import java.time.ZoneOffset;
+import java.util.List;
+import java.util.Map;
+import org.junit.jupiter.api.Test;
+
+public class OutlierAnalysisServiceTest {
+  private final DataSourceDefinitionAuthorizationService definitionAuthorizationService =
+      mock(DataSourceDefinitionAuthorizationService.class);
+  private final DurationOutliersReader outliersReader = mock(DurationOutliersReader.class);
+  private final OutlierAnalysisService outlierAnalysisService =
+      new OutlierAnalysisService(definitionAuthorizationService, outliersReader);
+
+  @Test
```

Comment:
The selection of tests here seems somewhat random. As an alternative, I would suggest:

- One parameterized test checking for when authorization passes. This should be parameterized with the different public methods of this class. The assertions should check that the parameter is sent to the corresponding `outliersReader` method. This means that the forwarded request must also be parameterized
- Another parameterized test checking for when authorization is missing. In this case, the method under test can be a parameter, but the assertion is always the same regardless

Reply:
@RomanJRW parameterized test will be bad solution here, because we have four different methods  with for 3 different models and with generic types. will be more clear create private method with deduplicating solution

Reply:
fixed